File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.go
func firstUniqChar(s string) int {
    stack := make(map[string]int)

    for i := 0; i < len(s); i++ {
        stack[string(s[i])]++
    }

    for i := 0; i < len(s); i++ {
        if stack[string(s[i])] == 1 {
            return i
        }
    }
    return -1
}

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        map<char, int>charCounts;
        for (const char c : s) {
            ++charCounts[c];
        }
        for (int i = 0; i < s.size(); ++i) {
            if (charCounts[s[i]] == 1) {
                return i;
            }
        }
        return -1;
    }
};